### PR TITLE
Last Recently Used #336 Update WaypointList.cpp

### LINKED ISSUE #336: InfoBox - Last Recently Used
Es wäre toll, wenn man eine Infobox LRU Wayponts hätte. Durch Tippen auf diese würde man dann eine Liste der letzten ausgewählten Waypoints sehen. Im Fluge wechselt man oft zwischen Waypoints hin und her. Das umschalten auf die LRU Waypoints ist aber etwas mühselig.

### PR DESCRIPTION
**Pull request by kxmm**
Change of the WaypointList.cpp so that ist shows the Last Waypoints when entering the dialog.
additionally there is a LRU "Letzte" Button to fetch the LRU after having queried other waypoints

### CODE REVIEW

**Comment by MaxKellermann** [comment]
Bad commit message. If your PR contains text, then your commit message is bad.

File: src/Dialogs/Waypoint/WaypointList.cpp [hunk 1]
```diff
@@ -238,6 +239,7 @@ class WaypointListButtons : public RowFormWidget {
   void Prepare(ContainerWindow &parent, const PixelRect &rc) override {
     AddButton(_("Select"), *list, SELECT);
     AddButton(_("Cancel"), dialog, mrCancel);
+    AddButton(_("Letzte"), *list, LRU);
```
Comment (MaxKellermann) [inline src/Dialogs/Waypoint/WaypointList.cpp:242]: Must be English.

**Comment by MaxKellermann** [comment]
This PR confuses me. Can you squash those fixup commits and write a proper commit message?

File: src/Dialogs/Waypoint/WaypointList.cpp [hunk 2]
```diff
@@ -314,11 +317,15 @@ WaypointListWidget::UpdateList()
   if (dialog_state.type_index == TypeFilter::LAST_USED)
     FillLastUsedList(items, LastUsedWaypoints::GetList(),
                      way_points);
-  else
-    FillList(items, way_points, location, last_heading,
-             dialog_state,
-             ordered_task, ordered_task_index);
-
+  else {
+      FillList(items, way_points, location, last_heading,
```
Comment (MaxKellermann) [inline src/Dialogs/Waypoint/WaypointList.cpp:321]: Bad indent

File: .gitignore [hunk 3]
```diff
@@ -33,3 +33,4 @@ build/local-config.mk
 *.pyo
 *.bak
 .*.swp
+/.vs
```
Comment (MaxKellermann) [inline .gitignore:36]: What's this, and why is it hidden inside this commit?

File: src/Dialogs/Waypoint/WaypointList.cpp [hunk 4]
```diff
@@ -314,11 +317,15 @@ WaypointListWidget::UpdateList()
...
+          dialog_state,
+          ordered_task, ordered_task_index);
+      if (items.size() == 0) {
```
Comment (MaxKellermann) [inline src/Dialogs/Waypoint/WaypointList.cpp:324]: Check `.empty()` which is a lot faster on some container types.

File: src/Dialogs/Waypoint/WaypointList.cpp [hunk 5]
```diff
@@ -314,11 +317,15 @@ WaypointListWidget::UpdateList()
...
+          FillLastUsedList(items, LastUsedWaypoints::GetList(),
+              way_points);
```
Comment (MaxKellermann) [inline src/Dialogs/Waypoint/WaypointList.cpp:326]: Bad indent

File: .gitignore (hunk 3 above)
Comment (kxmm) [reply .gitignore:36]: Visual Studio Files are ignored

Comment (MaxKellermann) [reply .gitignore:36]: Why?
And (repeating myself): why is it hidden inside this commit?

Comment (kxmm) [reply .gitignore:36]: Hallo Max, this was not on purpose and you do not have to take this change into the main branch. Do you want me to change it, or do you want to make the change yourself.

Comment (MaxKellermann) [reply .gitignore:36]: Amend the commit and remove it. This doesn't belong here.

**Comment by MaxKellermann** [comment]
This PR now contains a merge commit. This is wrong.
Also don't add fixup commits; amend the original commit if there are known mistakes.

File: src/Dialogs/Waypoint/WaypointList.cpp [hunk 6]
```diff
@@ -314,11 +317,14 @@ WaypointListWidget::UpdateList()
   if (dialog_state.type_index == TypeFilter::LAST_USED)
     FillLastUsedList(items, LastUsedWaypoints::GetList(),
                      way_points);
-  else
+  else {
     FillList(items, way_points, location, last_heading,
-             dialog_state,
-             ordered_task, ordered_task_index);
-
+             dialog_state, ordered_task, ordered_task_index);
```
Comment (MaxKellermann) [inline src/Dialogs/Waypoint/WaypointList.cpp:322]: Why this change?

**Comment by MaxKellermann** [comment]
I didn't see that you pushed a fix long ago, because you didn't mark my review as "resolved".
Your commit has a bad author name/email.

File: src/Dialogs/Waypoint/WaypointList.cpp (hunk 6 above)
Comment (kxmm) [reply src/Dialogs/Waypoint/WaypointList.cpp:322]: easier to read

**Review by kxmm — COMMENTED** [review]
Changed Mail and Name

**Comment by kxmm** [comment]
@MaxKellermann 
Hi Max,
just wanted to know if thereis something missing. Ohterwise it woudl be great if you could merge this fix.

File: src/Dialogs/Waypoint/WaypointList.cpp (hunk 6 above)
Comment (MaxKellermann) [reply src/Dialogs/Waypoint/WaypointList.cpp:322]: But this change is not relevant to this PR; it is confusing reviewers.

**Comment by MaxKellermann** [comment]
Your PR conflicts with the master branch because it's still based on a commit from last year. You need to rebase each time you push an updated PR branch.

**Comment by lordfolken** [comment]
@kxmm you need to rebase, as there is a conflict with changes that have happend in the waypointList file. 
https://medium.com/@dirk.avery/the-definitive-git-rebase-guide-dbd7717f9437 is a good guide.

**Comment by lordfolken** [comment]
@kxmm can you please rebase, so we can merge it?

**Comment by MaxKellermann** [comment]
Let's reopen when the problem is fixed.
